content/articles/understanding-and-implementing-graphs-in-java/index.md

```diff
@@ -0,0 +1,316 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /understanding-and-implementing-graphs-in-java/
+title: Understanding and implementing graphs in Java
+description: This tutorial will cover the types, implementation of, and basic concepts of a graph as a data structure in Java.
+author: teresia-mburu
+date: 2022-01-24T00:00:00-14:20
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/understanding-and-implementing-graphs-in-java/hero.jpg
+    alt: Understanding and implementing graphs in Java Hero Image
+---
+There are possibilities of storing information in a java data structure graph. The idea of a graph depicts a network that links several different points in time. 
+<!--more-->
+While covering graph data structure, we will learn about the various types of graphs, their implementation, and traversal over the graph. We will also cover the fundamental principles as a data structure, its implementation in java, and the numerous operations performed on a graph.
+
+### Table of contents
+- [Understanding graphs in java](#understanding-graphs-in-java)
+- [Graphs with various configurations](#graphs-with-various-configurations)
+- [Storing graphs in memory](#storing-graphs-in-memory)
+- [Representing graphs in code](#representing-graphs-in-code)
+- [Graph implementation in java](#graph-implementation-in-java)
+- [Common operations to implement in java](#common-operations-to-implement-in-java)
+- [Graph traversal methods and their implementation](#graph-traversal-methods-and-how-they-are-implemented)
+- [Depth-first traversal and its implementation](#depth-first-traversal-and-its-implementation)
+- [Breadth-first traversal and its implementation](#breadth-first-traversal-and-its-implementation)
+
+### Understanding graphs in java
+A graph data structure holds data related to a network of people or cities. It is made up of nodes known as vertices. The formation of the linkages is a connection of a vertex to another forming edge. 

```

Comment:
A graph data structure holds data related to a network of `people` or `cities`  -  is not limited to these two.